apis/pkg/v1alpha1/interfaces.go

```diff
@@ -240,6 +253,14 @@ func (p *Configuration) SetIgnoreCrossplaneConstraints(b *bool) {
 	p.Spec.IgnoreCrossplaneConstraints = b
 }
 
+// GetControllerConfigRef of this Configuration.
+func (p *Configuration) GetControllerConfigRef() *runtimev1alpha1.Reference {
+	return nil
+}
+
+// SetControllerConfigRef of this Configuration.
+func (p *Configuration) SetControllerConfigRef(r *runtimev1alpha1.Reference) {}
```

Comment:
Exposing these stub no-op getters and setters for a type that doesn't actually use (and may never use) a controller config feels like a bit of a code smell to me.

Reply:
I very much agree, but it does allow the package types to be handled generically. I do think it makes sense to just pass in "revision configurators" to the package manager controllers at this point, but I may defer that refactor to a separate PR if that sounds good.